### ISSUE TITLE
[expo-face-detector] using old version of GTMSessionFetcher/Core 

### ISSUE THREAD

**Issue by davoam** [expo/expo#25788]
### Minimal reproducible example

Run `pod install` with the packages version mentioned in the summary

### Summary

TL;DR `@react-native-firebase/auth` and `expo-face-detector` use the same dependency but because `expo-face-detector` use too old version of `GTMSessionFetcher/Core` the latest versions of the mentioned packages cannot be used in the same project

**Detailed explanation:** 

When `@react-native-firebase/auth` version `18.7.1` is used along with `expo-face-detector` version `12.4.0`,
the following error pops up
```
[!] CocoaPods could not find compatible versions for pod "GTMSessionFetcher/Core":
  In Podfile:
    EXFaceDetector (from `../node_modules/expo-face-detector/ios`) was resolved to 12.6.0, which depends on
      MLKitCommon (= 5.0.0) was resolved to 5.0.0, which depends on
        GTMSessionFetcher/Core (~> 1.1)

    RNFBAuth (from `../node_modules/@react-native-firebase/auth`) was resolved to 18.7.1, which depends on
      Firebase/Auth (= 10.18.0) was resolved to 10.18.0, which depends on
        FirebaseAuth (~> 10.18.0) was resolved to 10.18.0, which depends on
          GTMSessionFetcher/Core (< 4.0, >= 2.1)
```

It happens because `EXFaceDetector` depends on `MLKitCommon` version `5.0.0` (specified [here](https://github.com/expo/expo/blob/main/packages/expo-face-detector/ios/EXFaceDetector.podspec)) which depends on
`GTMSessionFetcher/Core` version `~> 1.1` (specified [here](https://github.com/CocoaPods/Specs/blob/master/Specs/c/c/6/MLKitCommon/5.0.0/MLKitCommon.podspec.json#L4C4-L6C7)).

`RNFBAuth` depends on `Firebase/Auth` which starting from version `10.0.0` depends on `GTMSessionFetcher/Core`, version `~> 2.1`
(specified [here](https://github.com/firebase/firebase-ios-sdk/blame/0dd5b06304a699444d84fc18a2fec9c41d9c6f7c/FirebaseFunctions.podspec#L46C3-L46C50))

So, `GTMSessionFetcher/Core` versions are incompatible.

It seems that `MLKitCommon` starting from version `9.0.0` (see this [file](https://github.com/CocoaPods/Specs/blob/master/Specs/c/c/6/MLKitCommon/9.0.0/MLKitCommon.podspec.json#L4C4-L6C7)) use newer versions of `GTMSessionFetcher/Core`

Related issue:
https://github.com/expo/expo/issues/20644

**Question**: is it possible to update version from `5.0.0` to `9.0.0`?

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 14.1.2
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.15.0 - ~/.nvm/versions/node/v18.15.0/bin/node
      npm: 9.5.0 - ~/.nvm/versions/node/v18.15.0/bin/npm
      Watchman: 2023.09.04.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.14.3 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.10121639
      Xcode: 15.0.1/15A507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^48.0.0 => 48.0.10 
      react: 18.2.0 => 18.2.0 
      react-native: 0.71.13 => 0.71.13 
    Expo Workflow: bare

**Comment by brentvatne** [comment]
unfortunately we've decided to deprecate expo-face-detector in sdk 50, and we intend to drop support for expo-face-detector in sdk 51. it's a bit too niche for a use case for us to maintain as part of the expo sdk. i'd recommend forking it out into a package that you control, where you can bump the version as you need. you may alternatively want to investigate using something like react-native-vision-camera - their frame processor plugin guide explains how to build a frame processor for face detection: https://react-native-vision-camera.com/docs/guides/frame-processors-plugins-ios
